File: src/backend/Khadamati.Domain/Entities/Service.cs
using Khadamati.Domain.Common;

namespace Khadamati.Domain.Entities;

public class Service : BaseEntity
{
    public Guid CategoryId { get; set; }
    public string NameAr { get; set; } = string.Empty;
    public string NameEn { get; set; } = string.Empty;
    public string? DescriptionAr { get; set; }
    public string? DescriptionEn { get; set; }
    public decimal BasePrice { get; set; }
    public string? ImageUrl { get; set; }
    public bool IsActive { get; set; } = true;
    public int EstimatedDurationMinutes { get; set; }

    public ServiceCategory Category { get; set; } = null!;
    public ICollection<CraftsmanService> CraftsmanServices { get; set; } = new List<CraftsmanService>();
    public ICollection<ServiceRequest> Requests { get; set; } = new List<ServiceRequest>();
}

File: src/ios/Khadamati/Core/Models/Service.swift
import Foundation

struct Service: Codable, Identifiable, Equatable {
    let id: UUID
    let categoryId: UUID
    let nameAr: String
    let nameEn: String
    let descriptionAr: String?
    let descriptionEn: String?
    let basePrice: Decimal
    let imageUrl: String?
    let estimatedDurationMinutes: Int

    func localizedName(locale: Locale = .current) -> String {
        locale.language.languageCode?.identifier == "ar" ? nameAr : nameEn
    }

    func localizedDescription(locale: Locale = .current) -> String? {
        locale.language.languageCode?.identifier == "ar" ? descriptionAr : descriptionEn
    }

    var formattedPrice: String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .currency
        formatter.currencyCode = "USD"
        formatter.locale = Locale(identifier: "en_LB")
        return formatter.string(from: basePrice as NSDecimalNumber) ?? "\(basePrice)"
    }
}
